sbt_common/common/src/main/scala/uk/ac/wellcome/WellcomeApp.scala

```diff
@@ -0,0 +1,35 @@
+ package uk.ac.wellcome
+
+ import akka.Done
+ import com.typesafe.config.{Config, ConfigFactory}
+ import grizzled.slf4j.Logging
+
+ import scala.concurrent.duration.Duration
+ import scala.concurrent.{Await, Future}
+
+ trait WorkerService {
+   def run(): Future[Done]
+ }
+
+ trait WellcomeApp extends App with Logging {
+   def buildWorkerService(config: Config): WorkerService
```

Comment:
can we avoid overriding this function and have `run` take a `WorkerService`?

Reply:
I was trying to ditch having this line in every app:

```scala
val config: Config = ConfigFactory.load()
```

How about if `run()` looks a bit like this:

```scala
def run(buildWorkerService: (config: Config) => WorkerService) {
    val config: Config = ConfigFactory.load()
    val workerService = buildWorkerService(config)
    workerService.run()
}
```

Would that work for you?

Reply:
Yeah ok. 
`config` is never used in `WellcomeApp` other than to build the `WorkerService ` so it feels to me like it shouldn't be the concern of `WellcomeApp` to instantiate it. The fact that it does feels a bit artificial and I don't think it's quite in the right place. 
But it's better than overriding, I would be happy to go with that for now